CMakePresets.json

```diff
@@ -1,19 +1,24 @@
 {
-    "version": 3,
-    "configurePresets": [
-        {
-            "name": "default",
-            "binaryDir": "build/${presetName}",
-            "cacheVariables": {
-                "BUILD_SAMPLES": true,
-                "CMAKE_INSTALL_PREFIX": "install/${presetName}"
-            }
-        }
-    ],
-    "buildPresets": [
-        {
-            "name":"default",
-            "configurePreset": "default"
-        }
-    ]
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "default",
+      "binaryDir": "build/${presetName}",
+      "cacheVariables": {
+        "BUILD_SAMPLES": true,
+        "CMAKE_INSTALL_PREFIX": "install/${presetName}"
+      }
+    },
+    {
+      "name": "vcpkg",
```

Comment:
This adds the `vcpkg` cmake preset, which will not require any dependencies if using Visual Studio's workspace (open folder) mode, but on other platforms/IDES would require vcpkg to be installed and `VCPKG_ROOT` env var to be set as per the cmake/vcpkg docs

An example of using this on the command line would be `cmake --preset vcpkg`. It will automatically install any required dependencies using vcpkg